tests/try_assert.py

```diff
@@ -0,0 +1,43 @@
+"""
+This test is executed separately due to pytest's assertion-rewriting
+"""
+import sys
+from typing import List
+
+from pydantic import BaseModel, ValidationError, validator
+
+
+def test_assert_raises_validation_error():
+    test_name = test_assert_raises_validation_error.__name__
+
+    class Model(BaseModel):
+        a: str
+
+        @validator('a')
+        def check_a(cls, v):
+            assert v == 'a', 'invalid a'
+            return v
+
+    Model(a='a')
+    expected_errors = [{'loc': ('a',), 'msg': f'invalid a', 'type': 'assertion_error'}]
+
+    try:
+        Model(a='snap')
+    except ValidationError as exc:
+        actual_errors = exc.errors()
+        if actual_errors != expected_errors:
+            description = [f'Actual errors: {actual_errors}', f'Expected errors: {expected_errors}']
+            fail_test(test_name, description)
+    else:
+        fail_test(test_name, ['ValidationError was not raised'])
```

Comment:
```suggestion
        raise RuntimeError(f'{test_name}: ValidationError was not raised')
```
and remove `fail_test`, if you really want to avoid duplication you could implement a custom exception with the logic from `fail_test`.

That way it reads better and the traceback is printed, with the traceback we don't really need to include the test name.